tests/lib/rules/component-tags-order.js

```diff
@@ -102,11 +104,90 @@ tester.run('component-tags-order', rule, {
       output: null,
       options: [{ order: ['docs', 'script', 'template', 'style'] }]
     },
+    {
+      code: '<script setup></script><script></script><template></template><style></style>',
+      output: null,
+      options: [{ order: ['script[setup]', 'script', 'template', 'style'] }]
```

Comment:
I think we should probably test `script:not([setup])`.

```suggestion
      options: [{ order: ['script[setup]', 'script:not([setup])', 'template', 'style'] }]
```